src/Client.php

```diff
@@ -25,10 +25,13 @@ class Client extends ElasticaClient
     private $indexer;
     private $indexBuilder;
 
+    /** @var AbstractConfiguration $configuration */
+    private $configuration;
```

Comment:
I thinks what you wanted to achieve here is more a "MappingReader" than a "Configuration"; configuration is still an array given to the Client and this new AbstractConfiguration class looks ambiguous.

Reply:
I do agree I updated in this way